frontend/src/metabase/visualizations/visualizations/PieChart/chart-definition.ts

```diff
@@ -0,0 +1,209 @@
+  checkRenderable: (
+    [
+      {
+        data: { rows },
+      },
+    ],
+    settings,
+  ) => {
+    // This prevents showing "Which columns do you want to use" when
+    // the piechart is displayed with no results in the dashboard
+    if (rows.length < 1) {
+      throw new MinRowsError(1, 0);
+    }
+    if (!settings["pie.dimension"] || !settings["pie.metric"]) {
+      throw new ChartSettingsError(t`Which columns do you want to use?`, {
+        section: `Data`,
+      });
+    }
+  },
+  placeholderSeries: [
+    {
+      card: {
+        display: "pie",
+        visualization_settings: { "pie.show_legend": false },
+        dataset_query: { type: "query" },
+      },
+      data: {
+        rows: [
+          ["Doohickey", 3976],
+          ["Gadget", 4939],
+          ["Gizmo", 4784],
+          ["Widget", 5061],
+        ],
+        cols: [
+          { name: "Category", base_type: "type/Category" },
+          { name: "Count", base_type: "type/Integer" },
+        ],
+      },
+    },
+  ] as RawSeries,
+  settings: {
+    ...columnSettings({ hidden: true }),
+    ...dimensionSetting("pie.dimension", {
+      section: t`Data`,
+      title: t`Dimension`,
+      showColumnSetting: true,
+    }),
+    ...metricSetting("pie.metric", {
+      section: t`Data`,
+      title: t`Measure`,
+      showColumnSetting: true,
+    }),
+    "pie.show_legend": {
+      section: t`Display`,
+      title: t`Show legend`,
+      widget: "toggle",
+      getDefault: getDefaultShowLegend,
+      inline: true,
+      marginBottom: "1rem",
+    },
+    "pie.show_total": {
+      section: t`Display`,
+      title: t`Show total`,
+      widget: "toggle",
+      getDefault: getDefaultShowTotal,
+      inline: true,
+    },
+    "pie.percent_visibility": {
+      section: t`Display`,
+      title: t`Show percentages`,
+      widget: "radio",
+      getDefault: getDefaultPercentVisibility,
+      props: {
+        options: [
+          { name: t`Off`, value: "off" },
+          { name: t`In legend`, value: "legend" },
+          { name: t`On the chart`, value: "inside" },
+        ],
+      },
+    },
+    "pie.slice_threshold": {
+      section: t`Display`,
+      title: t`Minimum slice percentage`,
+      widget: "number",
+      getDefault: getDefaultSliceThreshold,
+    },
+    "pie.colors": {
+      section: t`Display`,
+      title: t`Colors`,
+      widget: "colors",
+      getDefault: getDefaultColors,
+      getProps: (series, settings) => ({
+        seriesValues: settings["pie._dimensionValues"] || [],
+        seriesTitles: settings["pie._dimensionTitles"] || [],
+      }),
+      getDisabled: (series, settings) => !settings["pie._dimensionValues"],
+    },
+    // this setting recomputes color assignment using pie.colors as the existing
+    // assignments in case the user previous modified pie.colors and a new value
+    // has appeared. Not ideal because those color values will be missing in the
+    // settings UI
+    "pie._colors": {
```

Comment:
Also currently unused but I wrote another comment about keeping all colors logic here so maybe this will be used

Reply:
Will remove